### PR TITLE
VID fix & core unit tests + Regression ValueMap Producer code cleanup. (76X)

### PR DESCRIPTION
**Pull request by lgray**
Addresses some recent issues (and fixes one) for maintaining out-of-the-box functionality for VID.
Particularly that VID doesn't run on top of MiniAOD right now, after this PR it does.

There are now unit tests that ensure Electron/Photon VID runs on MiniAOD and AOD out of the box.
This will help ensure problems like the one above are not encountered again.

Fixes a small bug in the calculation of regression input values.

Vectors started with size at least three, then push_back was used to assign instead of the index.

Small changes expected to MiniAOD electron energy.

### CODE REVIEW

**Comment by lgray** [comment]
@ikrav I am working on the 74X backport now. You may use that for user recipes once it is available.

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
@matteosan1 Edits to regression valuemap producer.

**Comment by lgray** [comment]
@cmsbuild please test

File: RecoEgamma/ElectronIdentification/plugins/ElectronRegressionValueMapProducer.cc [hunk 1]
```diff
@@ -20,32 +20,190 @@
...
+    edm::Ptr<reco::CaloCluster> pclus;
+    if( !missing_clusters ) {
+      // loop over all clusters that aren't the seed  
+      auto clusend = the_sc->clustersEnd();
+      for( auto clus = the_sc->clustersBegin(); clus != clusend; ++clus ) {
+        pclus = *clus;
+      
+        if(theseed == pclus ) 
+          continue;
+        _clusterRawEnergy[iclus]  = pclus->energy();
```
Comment (slava77) [inline RecoEgamma/ElectronIdentification/plugins/ElectronRegressionValueMapProducer.cc]: note for posterity: this is the part of #11236, which actually changes the saved values; the rest is just refactoring

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/11229	8b65eb7

- changes in the code are OK, mostly refactoring and addition of unit tests. there is also a bugfix in the regresion input variable computations
- plot from wflow 4.53 in jenkins shows the expected small change in the miniAOD electron kinematics
wf4 53_pt_slimmedele
- additionally, tested locally in CMSSW_7_6_0_pre5 /test area sign590/ to check on run2 (run 251721) with more stats and check technical performance
    - plot from run 251721 DoubleEG 200 events shows about the same with a bit more statistics
wf251721doubleeg_pt_slimmedele
    - timing and event sizes are essentially unchanged
- use of isnan should be replaced with something else, as reported by the static analyzer, e.g.
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11229/8068/llvm-analysis/report-9c70ee.html#EndPath
Due to broken IB at this point, I suggest to make a fix in a separate technical PR.

**Comment by davidlange6** [comment]
+1

**Pull request merged**